core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1700,19 +1700,34 @@ class SparkContext(config: SparkConf) extends Logging {
    * Adds a JAR dependency for all tasks to be executed on this SparkContext in the future.
    * The `path` passed can be either a local file, a file in HDFS (or other Hadoop-supported
    * filesystems), an HTTP, HTTPS or FTP URI, or local:/path for a file on every worker node.
+   * If addToCurrentClassLoader is true, attempt to add the new class to the current threads' class
+   * loader. In general adding to the current threads' class loader will impact all other
+   * application threads unless they have explicitly changed their class loader.
    */
   def addJar(path: String) {
+    addJar(path, false)
+  }
+
+  def addJar(path: String, addToCurrentClassLoader: Boolean) {
```

Comment:
What about if we made the the Scala API as a Developer API since we might end up wanting to change the Scala side in the future?